standard_sequence/experiment_components.py

```diff
@@ -0,0 +1,816 @@
+        devices.octagon_456_aom_analog.constant(t, const)  # analog to const
+        self.blue_power = const
+
+    def blue_pulse_aom_off(self, t):
+        devices.octagon_456_aom_digital.go_low(t)  # digital off
+        devices.octagon_456_aom_analog.constant(t, 0)  # analog off
+        self.blue_power = 0
+
+    def red_servo_aom_on(self, t, const):
+        devices.ipg_1064_aom_digital.go_high(t)  # digital on
+        devices.ipg_1064_aom_analog.constant(t, const)  # analog to const
+        self.red_power = const
+
+    def red_servo_aom_off(self, t):
+        devices.ipg_1064_aom_digital.go_low(t)  # digital off
+        devices.ipg_1064_aom_analog.constant(t, 0)  # analog off
+        self.red_power = 0
+
+    def red_pulse_aom_on(self, t, const):
+        devices.pulse_1064_digital.go_high(t)  # digital on
+        devices.pulse_1064_analog.constant(t, const)  # analog to const
+        self.red_power = const
+
+    def red_pulse_aom_off(self, t):
+        devices.pulse_1064_digital.go_low(t)  # digital off
+        devices.pulse_1064_analog.constant(t, 0)  # analog off
+        self.red_power = 0
+
+    def blue_mirror_1_position(self, t):
+        devices.mirror_1_horizontal.constant(
+            t, shot_globals.ryd_456_mirror_1_h_position
+        )
+        devices.mirror_1_vertical.constant(t, shot_globals.ryd_456_mirror_1_v_position)
+
+    def blue_mirror_2_position(self, t):
+        devices.mirror_2_horizontal.constant(
+            t, shot_globals.ryd_456_mirror_2_h_position
+        )
+        devices.mirror_2_vertical.constant(t, shot_globals.ryd_456_mirror_2_v_position)
+
+    def red_mirror_1_position(self, t):
+        devices.mirror_3_horizontal.constant(
+            t, shot_globals.ryd_1064_mirror_1_h_position
+        )
+        devices.mirror_3_vertical.constant(t, shot_globals.ryd_1064_mirror_1_v_position)
+
+    def red_mirror_2_position(self, t):
+        devices.mirror_4_horizontal.constant(
+            t, shot_globals.ryd_1064_mirror_2_h_position
+        )
+        devices.mirror_4_vertical.constant(t, shot_globals.ryd_1064_mirror_2_v_position)
+
+
+class UVLamps:
+    def __init__(self, t):
+        # Turn off UV lamps
+        devices.uv_switch.go_low(t)
+
+    def uv_pulse(self, t, dur):
+        """Flash the UV LED lamps for dur seconds"""
+        devices.uv_switch.go_high(t)
+        t += dur
+        devices.uv_switch.go_low(t)
+        return t
+
+
+class BField:
+    CONST_COIL_OFF_TIME: ClassVar[float] = (
+        1.4e-3  # minimum time for the MOT coil to be off
+    )
+    CONST_COIL_RAMP_TIME: ClassVar[float] = (
+        100e-6  # ramp time from initial field to final field
+    )
+    CONST_BIPOLAR_COIL_FLIP_TIME: ClassVar[float] = (
+        10e-3  # the time takes to flip the polarity of the coil
+    )
+    CONST_COIL_FEEDBACK_OFF_TIME: ClassVar[float] = (
+        4.5e-3  # how long to turn off the feedback of circuit when flipping polarity
+    )

```

Comment:
formatting, place comments before the variable to avoid the split line definition

Reply:
corrected